ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryReplica.java

```diff
@@ -103,6 +105,32 @@ public boolean isSealed() {
     return isSealed;
   }
 
+  @Override
+  public JSONObject getSnapshot() {
+    JSONObject snapshot = new JSONObject();
+    DataNodeId dataNodeId = getDataNodeId();
+    snapshot.put(REPLICA_NODE, dataNodeId.getHostname() + ":" + dataNodeId.getPort());
+    snapshot.put(REPLICA_PARTITION, getPartitionId().toPathString());
+    snapshot.put(REPLICA_DISK, getDiskId().getMountPath());
+    snapshot.put(REPLICA_PATH, getReplicaPath());
+    snapshot.put(CAPACITY_BYTES, getCapacityInBytes());
+    snapshot.put(REPLICA_WRITE_STATE, isSealed() ? PartitionState.READ_ONLY.name() : PartitionState.READ_WRITE.name());
+    String replicaLiveness = UP;
+    if (dataNodeId.getState() == HardwareState.UNAVAILABLE) {
```

Comment:
is this the precedence order followed by the router/ambryrequests?

Reply:
It looks to be so - it seems to be precedence order of `isDown()`